test-suite/js/src/test/scala/org/scalajs/testsuite/library/BigIntTest.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.library
+
+import org.junit.Assert._
+import org.junit.Assume._
+import org.junit.{BeforeClass, Test}
+import org.scalajs.testsuite.utils.Platform
+
+import scala.scalajs.js
+
+object BigIntTest {
+  @BeforeClass
+  def assumeRuntimeSupportsBigInt(): Unit = {
+    assumeTrue(Platform.executingInNodeJS)
+    val bigIntConstructor =
+      js.Dynamic.global.BigInt.asInstanceOf[js.UndefOr[js.Any]]
+    assumeTrue(bigIntConstructor.isDefined)
+  }
+}
+
+class BigIntTest {
+
+  @Test def apply(): Unit = {
+    val fromString = js.BigInt("9007199254740992")
+    assertEquals(fromString.toString(), "9007199254740992")
+
+    val fromInt = js.BigInt(2147483647)
+    assertEquals(fromInt.toString(), "2147483647")
+
+    val fromDouble = js.BigInt(4294967295d)
+    assertEquals(fromDouble.toString(), "4294967295")
+  }
+
+  @Test def asIntN(): Unit = {
+    val x = js.BigInt.asIntN(8, js.BigInt("256"))
+    assertEquals(x, js.BigInt("0"))
+  }
+
+  @Test def asUintN(): Unit = {
+    val x = js.BigInt(-123)
+    assertEquals(js.BigInt.asUintN(50, x), js.BigInt("1125899906842501"))
+  }
+
+  @Test def toLocaleString(): Unit = {
+    assumeTrue(Platform.executingInNodeJS)
+
+    val bi = js.BigInt("42123456789123456789")
+    assertEquals(bi.toString(), "42123456789123456789")
+
+    val currency = bi
+      .toLocaleString("de-DE", new js.BigInt.ToLocaleStringOptions {
+        style = "currency"
+        currency = "EUR"
+      })
+    // Internationalization support is limited by default in Node.js.
+    // https://nodejs.org/api/intl.html#intl_internationalization_support
+    // toLocaleString may return as same as by toString
+    assertTrue(
+      currency == "€ 42,123,456,789,123,456,789.00" ||
+      currency == "42123456789123456789")
```

Comment:
Sorry, I probably wasn't very clear the last time. Those two lines should be indented 4 spaces (2 more spaces than now).

Reply:
Fixed in https://github.com/scala-js/scala-js/pull/3786/commits/91fd0a6509b6ad566a27a270815bf7e360f979a6